src/WingetCreateCore/Models/Manifests.cs

```diff
@@ -40,6 +40,40 @@ public class Manifests
         /// </summary>
         public List<LocaleManifest> LocaleManifests { get; set; } = new List<LocaleManifest>();
 
+        /// <summary>
+        /// Generates the proper file name for the given manifest.
+        /// </summary>
+        /// <typeparam name="T">Manifest type.</typeparam>
+        /// <param name="manifest">Manifest object model.</param>
+        /// <returns>File name string of manifest.</returns>
+        public static string GetFileName<T>(T manifest)
+        {
+            string fileName = null;
+
+            if (manifest is InstallerManifest installerManifest)
+            {
+                fileName = $"{installerManifest.PackageIdentifier}.installer.yaml";
+            }
+            else if (manifest is VersionManifest versionManifest)
+            {
+                fileName = $"{versionManifest.PackageIdentifier}.yaml";
+            }
+            else if (manifest is DefaultLocaleManifest defaultLocaleManifest)
+            {
+                fileName = $"{defaultLocaleManifest.PackageIdentifier}.locale.{defaultLocaleManifest.PackageLocale}.yaml";
+            }
+            else if (manifest is LocaleManifest localeManifest)
+            {
+                fileName = $"{localeManifest.PackageIdentifier}.locale.{localeManifest.PackageLocale}.yaml";
+            }
+            else if (manifest is SingletonManifest singletonManifest)
+            {
+                fileName = $"{singletonManifest.PackageIdentifier}.yaml";
+            }
+
+            return fileName;
```

Comment:
You can use a switch expression here:

            return manifest switch
            {
                InstallerManifest installerManifest => $"{installerManifest.PackageIdentifier}.installer.yaml",
                VersionManifest versionManifest => $"{versionManifest.PackageIdentifier}.yaml",
                DefaultLocaleManifest defaultLocaleManifest => $"{defaultLocaleManifest.PackageIdentifier}.locale.{defaultLocaleManifest.PackageLocale}.yaml",
                LocaleManifest localeManifest => $"{localeManifest.PackageIdentifier}.locale.{localeManifest.PackageLocale}.yaml",
                SingletonManifest singletonManifest => $"{singletonManifest.PackageIdentifier}.yaml",
                _ => throw new ArgumentException(nameof(manifest)),
            };

Reply:
neat, changed to suggested. Thanks for the help :)

Reply:
Nice. Can you change basecommand.cs to use this method in the two places where we write files out too?

Reply:
Changed basecommand to use this method